src/QuokkaSimulation.hpp

```diff
@@ -1652,8 +1656,27 @@ void QuokkaSimulation<problem_t>::subcycleRadiationAtLevel(int lev, amrex::Real
 			auto const &prob_lo = geom[lev].ProbLoArray();
 			auto const &prob_hi = geom[lev].ProbHiArray();
 			// update state_new_cc_[lev] in place (updates both radiation and hydro vars)
-			operatorSplitSourceTerms(stateNew, indexRange, time_subcycle, dt_radiation, 2, dx, prob_lo, prob_hi, p_num_failed_coupling,
-						 p_num_failed_dust, p_num_failed_outer);
+			operatorSplitSourceTerms(stateNew, indexRange, time_subcycle, dt_radiation, 2, dx, prob_lo, prob_hi, p_iteration_counter,
+						 p_num_failed_coupling, p_num_failed_dust, p_num_failed_outer);
+		}
+
+		if (print_rad_counter_) {
+			auto h_iteration_counter = iteration_counter.copyToHost();
+			long global_solver_count = h_iteration_counter[0];  // number of Newton-Raphson solvings
```

Comment:
This is just the number of cells, right?

Reply:
No, because there is an outer iteration to update the work term. If there is always only one outer iteration, `global_solver_count` would be 2x number of cells (2 comes from two IMEX stages). Therefore, `global_solver_count` also counts the average number of outer iterations per radiation update.